src/content/docs/en/upgrade-astro.mdx

```diff
@@ -0,0 +1,188 @@
+
+Minor releases include changes such as:
+
+- **Deprecations** of existing features/options.
+- Introduction of new functionalities.
+- Upgrade of minimum required Node.js version (minor or patch), e.g. from `v18.4.0` to `v18.7.0`.
+  :::note
+  We take these kind of changes very seriously, and approve them only if we think that a new version of Node.js can improve the security or developer experience of our users, and impact core functionalities.
+  :::
+- Introduction of new options in the integration hooks.
+- Introduction of new functionalities in `astro/app`, notably used for creating new adapters.
+
+A minor release may also include smaller, patch changes at the same time.
+
+### Major changes
+
+Major releases will include breaking changes to at least some existing code. These breaking changes are always documented in an "Upgrade to vX" guide in Astro.
+
+Major releases allow Astro to make significant changes not only to internal logic, but also in intended behavior and usage. Documentation will be updated to reflect the latest version only, and a **static, unmaintained snapshot of older docs** are available as a historical record for older projects that are not yet upgraded.
+
+Major releases include changes such as:
+
+- Upgrade of minimum required Node.js version (minor or major), e.g. from `v18.4.0` to `v20.0.0`.
+- Removal of previously deprecated functionalities.
+- Changes of existing functionalities.
+- Changes of existing options in the integration hooks.
+- Changes of existing options and functionalities in `astro/app`, notably used for creating new adapters.
+
+A major release may also include some non-breaking changes and improvements that would normally be released separately in a minor or patch release.
+
+### Exceptions
+
+- **Experimental features**. This allows Astro developers the greatest flexibility to explore, and even radically change course, during development. Therefore, the behaviour of these features can break in minor and patch changes.
+
+  These features are usually accompanied by ongoing, public [Request for Consideration (RFC) stage 3](https://github.com/withastro/roadmap#stage-3-rfc--development). It is expected that beta users will follow for updates, and leave early feedback to help guide development of these features.
+
+  Once these features are out of their experimentation period, they will follow the normal semantic versioning contract.
+
+- **Improvements to the documentation** (e.g. reference and error messages). They are built from source from the `docs` repository. This allows Astro to quickly update docs fixes and improvements in the cases where documentation source content is stored in the main `astro` repo.
```

Comment:
Ensuring consistency.
```suggestion
- **Improvements to the documentation** (e.g. reference and error messages). They are built from source from the `docs` repository. This allows Astro to quickly update docs fixes and improvements in the cases where documentation source content is stored in the main `astro` repository.
```

Reply:
I will note that often we *intentionally* switch between equivalent words for sentence variety which helps for readability! Also, sometimes a shorter version of a word works better for rhythm in longer sentences and vice versa. 

No problems changing here, but just pointing out that 100% consistency isn't *always* the top priority!

Reply:
Sounds good! Will keep this in mind moving forward :)!